bigquery/test/tables.test.js

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var proxyquire = require('proxyquire').noCallThru();
+
+function getSample () {
+  var bucketMock = {
+    file: sinon.stub().returns(fileMock)
+  };
+
+  var storageMock = {
+    bucket: sinon.stub().returns(bucketMock)
+  };
+
+  var fileMock = {};
+
+  var metadataMock = { status: { state: 'DONE' } };
+
+  var jobId = 'abc';
+
+  var jobMock = {
+    id: jobId,
+    getMetadata: sinon.stub().callsArgWith(0, null, metadataMock)
+  };
+
+  var tableMock = {
+    export: sinon.stub().callsArgWith(2, null, jobMock)
+  };
+
+  var datasetMock = {
+    table: sinon.stub().returns(tableMock)
+  };
+
+  var bigqueryMock = {
+    job: sinon.stub().returns(jobMock),
+    dataset: sinon.stub().returns(datasetMock)
+  };
+
+  var BigQueryMock = sinon.stub().returns(bigqueryMock);
+  var StorageMock = sinon.stub().returns(storageMock);
+
+  return {
+    program: proxyquire('../tables', {
+      '@google-cloud/bigquery': BigQueryMock,
+      '@google-cloud/storage': StorageMock,
+      yargs: proxyquire('yargs', {})
+    }),
+    mocks: {
+      BigQuery: BigQueryMock,
+      bigquery: bigqueryMock,
+      Storage: StorageMock,
+      storage: storageMock,
+      metadata: metadataMock,
+      job: jobMock,
+      table: tableMock,
+      bucket: bucketMock,
+      dataset: datasetMock
+    },
+    jobId: jobId
```

Comment:
One benefit of doing it this way is that everything related to dummy values is in a single package - but I'll change over to your way for consistency with the other samples.